src/promisify.js

```diff
@@ -0,0 +1,142 @@
+/***************************************************************************************************
+
+The promisify implementation found in this file is derived from the implementation in Node.js. Its
```

Comment:
Hm that's a thought, but it is a significant departure, so this is just for licensing.